src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -41,49 +46,51 @@ public static RuntimeAssemblyName Parse(string s)
             string name;
             AssemblyNameLexer.Token token = lexer.GetNext(out name);
             if (token != AssemblyNameLexer.Token.String)
-                throw new FileLoadException(SR.InvalidAssemblyName);
+                throw new FileLoadException(SR.Argument_InvalidAssemblyName);
 
             if (name == string.Empty || name.IndexOfAny(s_illegalCharactersInSimpleName) != -1)
-                throw new FileLoadException(SR.InvalidAssemblyName);
+                throw new FileLoadException(SR.Argument_InvalidAssemblyName);
 
             Version? version = null;
             string? cultureName = null;
             byte[]? pkt = null;
             AssemblyNameFlags flags = 0;
 
-            LowLevelList<string> alreadySeen = new LowLevelList<string>();
+            List<string> alreadySeen = new List<string>();
             token = lexer.GetNext();
             while (token != AssemblyNameLexer.Token.End)
             {
                 if (token != AssemblyNameLexer.Token.Comma)
-                    throw new FileLoadException(SR.InvalidAssemblyName);
+                    throw new FileLoadException(SR.Argument_InvalidAssemblyName);
                 string attributeName;
 
                 token = lexer.GetNext(out attributeName);
                 if (token != AssemblyNameLexer.Token.String)
-                    throw new FileLoadException(SR.InvalidAssemblyName);
+                    throw new FileLoadException(SR.Argument_InvalidAssemblyName);
                 token = lexer.GetNext();
 
+#if CORERT
```

Comment:
If you are running into behavior differences between CoreCLR and CoreRT, they are CoreRT bugs. It is fine to switch CoreRT to CoreCLR behavior, no need for ifdefs.

Reply:
I suspected that desctop behavior should win here. I mostly just did not want to silently change the behavior in what is technically a refactoring change. Not without this kind of feedback on whether we want to change.